Subject: [PATCH] Fix potential heap memory leakage.

If failed to get SkMallocPixelRef, previous heap allocaion
should be freed.

diff --git a/core/jni/android/graphics/Graphics.cpp b/core/jni/android/graphics/Graphics.cpp
index 528541d..d36de69 100644
--- a/core/jni/android/graphics/Graphics.cpp
+++ b/core/jni/android/graphics/Graphics.cpp
@@ -562,6 +562,7 @@
                                                          &allocatePixelsReleaseProc, context);
     if (!pr) {
         delete context;
+        sk_free(addr);
         return false;
     }
 
